src/main/client/app/pages/modules/terraform-image-input.vue

```diff
@@ -0,0 +1,135 @@
+<template>
+  <b-form-group
+    label="Terraform Image"
+    :state="isFormValid"
+  >
+    <b-input-group>
+      <b-input-group-prepend is-text>
+        <b-form-checkbox
+          id="image_override"
+          v-model="isCustomImage"
+          switch
+          @change="overrideImage"
+        >
+          Custom
+        </b-form-checkbox>
+      </b-input-group-prepend>
+
+      <form-typeahead
+        v-if="isCustomImage"
+        id="image_repository"
+        v-model="image.repository"
+        placeholder="Repository"
+        :state="isRepositoryNotEmpty && isRepositoryValid"
+        :formatter="formatRepository"
+        group-position="middle"
+        source="/api/docker/repositories?name=%REPO"
+        wildcard="%REPO"
+        filter-property="name"
+        @input="resetTag"
+      />
+
+      <form-typeahead
+        id="image_tag"
+        v-model="image.tag"
+        placeholder="Tag"
+        :state="isTagNotEmpty"
+        :disabled="!isRepositoryNotEmpty || !isRepositoryValid"
+        :source="'/api/docker/repositories/' + image.repository + '/tags?name=%TAG'"
+        group-position="right"
+        wildcard="%TAG"
+        filter-property="name"
+        :min-length="1"
+      />
+    </b-input-group>
+
+    <b-form-text>
+      Default from <strong>hashicorp/terraform</strong>. Switch to override it
+    </b-form-text>
+
+    <template slot="invalid-feedback">
+      {{ invalidFeedback }}
+    </template>
+  </b-form-group>
+</template>
+
+<script>
+  import $ from 'jquery';
+
+  import { AppFormTypeahead as FormTypeahead } from '@/shared/components';
+
+  export default {
+    name: 'AppTerraformImageInput',
+
+    components: { FormTypeahead },
+
+    props: {
+      image: {
+        type: Object,
+        required: true,
+      },
+    },
+
+    data: () => ({
+      isCustomImage: null,
+      officialRepository: 'hashicorp/terraform',
+    }),
+
+    computed: {
+      isFormValid() {
+        const result = this.isRepositoryNotEmpty && this.isRepositoryValid && this.isTagNotEmpty;
+        // emit form status to parent
+        this.$emit('form-status', result);
+        this.$emit('input', this.image);
```

Comment:
why emitting the input too here ? we are not using binding with v-model :thinking: